From 225e7ea75f7d34fa742c12cf1245f817388471b5 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 13 Feb 2023 19:11:28 +0200
Subject: [PATCH] [SW-120178] fwupd: preserve boot_info on reset

While executing the device reset command, preserve the important
boot_info data.

Change-Id: I5ae6827309b78e0768e1aeb9210b42ca7041d904
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index ab9e1ca..9601327 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -17,6 +17,7 @@
 #include <fit_load_type.h>
 #include <security.h>
 #include <memory.h>
+#include <boot_info.h>
 
 #include "cmd_module.h"
 #include "uart.h"
@@ -771,8 +772,16 @@
 	__module_fwupd_run(&i2c_state);
 }
 
+static void module_fwupd_preserve_boot_info(void)
+{
+	preboot_finalize_fw_info();
+	preboot_finalize_nic_en();
+	boot_info_finalize();
+}
+
 static void module_fwupd_run_noret_reset(void)
 {
+	module_fwupd_preserve_boot_info();
 	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
 	hl_dev_reset();	/* should not return */
 }
